_admin_guides/ecr_registry.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: Configuring an ECR as a registry
+order: 120
+---
+
+This document reviews configuring ECR as a registry for a Spinnaker installation
+
+### Adding ECR as a Docker Registry
+
+When configuring a registry, you normally use the `hal` command for [adding a Docker Registry](https://www.spinnaker.io/reference/halyard/commands/#hal-config-provider-docker-registry-account-add).
+
+This works great for Dockerhub, but ECR requires a bit more work for configuration. Amazon ECR requires access tokens to access the images and those access tokens expire after a time.
+
+In order to automate updating the token, we will use a sidecar container with a script that does it for us. Since both Clouddriver and the sidecar container both need access to the ECR access token, we will use a shared volume to store the access token.
+
+The sidecar we're going to add does not start with an access token, it needs to be able to request an access token from ECR. The Spinnaker installation must have the `AmazonEC2ContainerRegistryReadOnly` policy attached to the role assigned in order to request and update the required access token.
+
+
+## Update Configs
+
+### Add a Sidecar for Token Refresh
+
+In your `~/.hal/config`, update the deploymentEnvironment.sidecars section:
+```
+  deploymentEnvironment:
+    sidecars:
+      spin-clouddriver:
+      - name: token-refresh
+        dockerImage: quay.io/skuid/ecr-token-refresh:latest
+        mountPath: /etc/passwords
+        configMapVolumeMounts:
+        - configMapName: token-refresh-config
+          mountPath: /opt/config/ecr-token-refresh
+```
+
+### Define ECR registry
+
+Create `~/.hal/<deployment>/profiles/clouddriver-local.yml`:
+```
+dockerRegistry:
+  enabled: true
+  accounts:
+  - name: my-ecr-registry
+    address: https://<aws-account-id>.dkr.ecr.<aws-region>.amazonaws.com
+    username: AWS
+    passwordFile: /etc/passwords/my-ecr-registry.pass
+```
+
+
+### Update your Spinnaker installation
+```
+hal deploy apply --service-names clouddriver
+```
+
+
+
+Create a token-refresh.yml
+
+```
+interval: 30m # defines refresh interval
+registries: # list of registries to refresh
+  - registryId: "<aws-account-id>"
+    region: "<aws-region>"
+    passwordFile: "/etc/passwords/my-ecr-registry.pass"
+```
+
+Apply it to the cluster with:
+```
+kubectl -n <namespace> create configmap token-refresh-config --from-file <token-refresh.yml location>
+```
+
+Now you can add ECR as a docker registry in the configuration stage
+
+![](https://d2ddoduugvun08.cloudfront.net/items/430k0r1Q2s1m1f032z3E/Image%202018-12-18%20at%202.02.02%20PM.png)
```

Comment:
Im not sure this is the correct way of adding an image, eg https://github.com/armory/documentation/blob/master/_install_guide/authz.md seems to use `cl.ly` as the domain, but the images resolve to `cloudfront.net` similar to what I have here.